doc/src/extras.rst

```diff
@@ -141,6 +141,383 @@ Logging cursor
+        log_conn = psycopg2.connect(dsn, connection_factory=LogicalReplicationConnection)
+        log_cur = log_conn.cursor()
+
+
+.. autoclass:: PhysicalReplicationConnection
+
+    This connection factory class can be used to open a special type of
+    connection that is used for physical replication.
+
+    Example::
+
+        from psycopg2.extras import PhysicalReplicationConnection
+        phys_conn = psycopg2.connect(dsn, connection_factory=PhysicalReplicationConnection)
+        phys_cur = phys_conn.cursor()
+
+    Both `LogicalReplicationConnection` and `PhysicalReplicationConnection` use
+    `ReplicationCursor` for actual communication on the connection.
+
+.. seealso::
+
+   - PostgreSQL `Streaming Replication Protocol`__
+
+   .. __: http://www.postgresql.org/docs/current/static/protocol-replication.html
+
+
+The individual messages in the replication stream are presented by
+`ReplicationMessage` objects:
+
+.. autoclass:: ReplicationMessage
+
+    .. attribute:: payload
+
+        The actual data received from the server.  An instance of either
+        ``str`` or ``unicode``, depending on the method that was used to
+        produce this message.
+
+    .. attribute:: data_size
+
+        The raw size of the message payload (before possible unicode
+        conversion).
+
+    .. attribute:: data_start
+
+        LSN position of the start of the message.
+
+    .. attribute:: wal_end
+
+        LSN position of the current end of WAL on the server.
+
+    .. attribute:: send_time
+
+        A `~datetime` object representing the server timestamp at the moment
+        when the message was sent.
+
+    .. attribute:: cursor
+
+        A reference to the corresponding `ReplicationCursor` object.
+
+
+.. autoclass:: ReplicationCursor
+
+    .. method:: identify_system()
+
+        Execute ``IDENTIFY_SYSTEM`` command of the streaming replication
+        protocol and return the result as a dictionary.
+
+        Example::
+
+            >>> cur.identify_system()
+            {'timeline': 1, 'systemid': '1234567890123456789', 'dbname': 'test', 'xlogpos': '0/1ABCDEF'}
+
+    .. method:: create_replication_slot(slot_name, output_plugin=None)
+
+        Create streaming replication slot.
+
+        :param slot_name: name of the replication slot to be created
+        :param slot_type: type of replication: should be either
+                          `REPLICATION_LOGICAL` or `REPLICATION_PHYSICAL`
+        :param output_plugin: name of the logical decoding output plugin to be
+                              used by the slot; required for logical
+                              replication connections, disallowed for physical
+
+        Example::
+
+            log_cur.create_replication_slot("logical1", "test_decoding")
+            phys_cur.create_replication_slot("physical1")
+
+            # either logical or physical replication connection
+            cur.create_replication_slot("slot1", slot_type=REPLICATION_LOGICAL)
```

Comment:
The signature documented for create_replication_slot above lacks slot_type argument, though one is present in the help and in the actual function definition:

```
Help on ReplicationCursor in module psycopg2.extras object:

class ReplicationCursor(psycopg2.extensions.cursor)
 |  A cursor used for communication on the replication protocol.
 |  
...
 |  create_replication_slot(self, slot_name, slot_type=None, output_plugin=None)
 |      Create streaming replication slot.
```

If the `LogicalReplicationConnection` and `PhysicalReplicationConnection` separation exists then this param shouldn't need to exist.

However, arguably that separation shouldn't be `LogicalReplicationConnection` and `PhysicalReplicationConnection` but rather `ReplicationConnection` and `DatabaseReplicationConnection`. Currently `replication=database` isn't used for anything else, but it might well be in future. Then `create_replication_slot` could take a `slot_type` argument, with either value legal for a `DatabaseReplicationConnection` and with only `REPLICATION_PHYSICAL` legal for a base `ReplicationConnection`.

Reasonable? Or am I over-complicating things? It seems like this would be a closer mapping to the underlying PostgreSQL API.